cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDatasetTest.java

```diff
@@ -43,7 +44,13 @@
   private static final Id.DatasetInstance datasetInstance =
     Id.DatasetInstance.from(DatasetFrameworkTestUtil.NAMESPACE_ID, "meta");
 
-  BusinessMetadataDataset dataset;
+  private BusinessMetadataDataset dataset;
+
+  private final Id.Application app1 = Id.Application.from("ns1", "app1");
+  // Have to use Id.Program here because Id.Flow does not implement equals. It needs to be updated.
```

Comment:
This comment isn't entirely correct. The real problem is that we do an ``equals`` comparison on lines 175 and 211. However, the search API that the result is from, actually returns an ``Id.Program``, not specifically an ``Id.Flow``. That's so the actual object is an ``Id.Program``, whereas the expected object If I used ``Id.Flow`` here would be ``Id.Flow``. That's the reason why the comparison fails. So unless the dataset APIs are modified to return the individual program Id classes (not worth the trouble IMO right now), we will have to use ``Id.Program`` here. I'll update the comment.